src/block/MonsterSpawner.php

```diff
@@ -38,10 +42,30 @@ protected function getXpDropAmount() : int{
 	}
 
 	public function onScheduledUpdate() : void{
-		//TODO
+		$world = $this->position->getWorld();
+		$spawner = $world->getTile($this->position);
+		if($spawner instanceof TileSpawner && $spawner->onUpdate()){
+			$world->scheduleDelayedBlockUpdate($this->position, 1);
+		}
 	}
 
 	public function getSupportType(int $facing) : SupportType{
 		return SupportType::NONE;
 	}
+
+	public function onInteract(Item $item, int $face, Vector3 $clickVector, ?Player $player = null, array &$returnedItems = []) : bool{
+		if($item instanceof SpawnEgg){
+			if($item->getEntityTypeId() === null){
+				return false;
+			}
+			$spawner = $this->position->getWorld()->getTile($this->position);
+			if($spawner instanceof TileSpawner){
+				$spawner->setEntityTypeId($item->getEntityTypeId());
```

Comment:
I think its better to store the egg in the spawner instead of the entityTypeId ? So, it can be used to create an entity later ?

Reply:
What do you mean store the egg in the spawner? Vanilla only saves the entity id

Reply:
The runtime implementation should not be dealing with entity IDs